commit 7b68e6271c0d65aaf13b11f96df43d8572db22fb
Author: Jerry Jelinek <jerry.jelinek@joyent.com>
Date:   2019-07-03T19:28:59+00:00 (3 months ago)
    
    OS-7879 zfs recv is broken
    Reviewed by: Kody Kantor <kody.kantor@joyent.com>
    Reviewed by: Jason King <jason.king@joyent.com>
    Approved by: Kody Kantor <kody.kantor@joyent.com>

diff --git a/usr/src/lib/libzfs/common/libzfs_sendrecv.c b/usr/src/lib/libzfs/common/libzfs_sendrecv.c
index 27f05f101e..2bf7c160de 100644
--- a/usr/src/lib/libzfs/common/libzfs_sendrecv.c
+++ b/usr/src/lib/libzfs/common/libzfs_sendrecv.c
@@ -3754,7 +3754,6 @@ zfs_receive_one(libzfs_handle_t *hdl, int infd, const char *tosnap,
 		}
 
 		newfs = B_TRUE;
-		*cp = '/';
 	}
 
 	zc.zc_begin_record = *drr_noswap;
